### PR TITLE
feat(plugin): Math/LaTeX Renderer

### PR DESCRIPTION
**Pull request by jumpyjacko**
Send some LaTeX images without a bot. Uses a web request to https://latex.codecogs.com/ to render and then canvas to invert colours.

Example
latex_example

Might be a better way to send it but I don't know how, I mainly copied /petpet.

### CODE REVIEW

**Comment by JumpyJacko** [comment]
Might have to change the LaTeX renderer API now that I'm double-checking [terms of usage](https://editor.codecogs.com/support/4-terms_of_use.php) and [fair usage](https://editor.codecogs.com/support/3-fair_usage.php). Need someone to help me out here.

**Comment by UDIA-GOD** [comment]
God damn thats crazy i need that

**Comment by CREAsTIVE** [comment]
i also wanna this too

**Comment by Curve** [comment]
You could use [MathJax](https://github.com/mathjax/MathJax-demos-node#MathJax-demos-node) to render on the client instead of a third-party API

**Comment by jumpyjacko** [comment]
> You could use [MathJax](https://github.com/mathjax/MathJax-demos-node#MathJax-demos-node) to render on the client instead of a third-party API

Yeah, my only concern really is whether its worth importing an entire Math rendering pacjage just for the one plugin that most people wouldn't use. I might move this plugin to community plugins instead.

File: src/plugins/mathRenderer.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import {
+    ApplicationCommandInputType,
+    ApplicationCommandOptionType,
+    findOption,
+    sendBotMessage,
+} from "@api/Commands";
+import { Devs } from "@utils/constants";
+import { findByCodeLazy } from "@webpack";
+
+const promptToUpload = findByCodeLazy("UPLOAD_FILE_LIMIT_ERROR");
```
Comment (AutumnVN) [inline src/plugins/mathRenderer.ts:29]: use UploadHandler from webpack/common

**Comment by jumpyjacko** [comment]
Closing, moving to user plugins on the Discord server.

**Comment by javalsai** [comment]
> Closing, moving to user plugins on the Discord server.

@jumpyjacko would you mind linking to the plugin? I'm really interested

**Comment by jumpyjacko** [comment]
> > Closing, moving to user plugins on the Discord server.
> 
> @jumpyjacko would you mind linking to the plugin? I'm really interested

I realised I had to rewrite but I couldn't figure out how to because my Javascript/Typescript is not the best so the plugin doesn't exist, sorry for the inconvenience. I also no longer use Vencord.

**Comment by javalsai** [comment]
> > > Closing, moving to user plugins on the Discord server.
> > 
> > 
> > @jumpyjacko would you mind linking to the plugin? I'm really interested
> 
> I realised I had to rewrite but I couldn't figure out how to because my Javascript/Typescript is not the best so the plugin doesn't exist, sorry for the inconvenience. I also no longer use Vencord.

:(
It's fine tho, I'll maybe end up writing myself a quick one when I get some time. Would really like to somehow manage to get autocompletion or view the rendered result as I've just learned LaTeX :P. I think discord should support LaTeX out of the box tho, the same way they have great support for codeblocks or file preview with syntax highlightning...

**Comment by jumpyjacko** [comment]
> :(
> It's fine tho, I'll maybe end up writing myself a quick one when I get some time. Would really like to somehow manage to get autocompletion or view the rendered result as I've just learned LaTeX :P. I think discord should support LaTeX out of the box tho, the same way they have great support for codeblocks or file preview with syntax highlightning...

You should be able to reuse most of my code for the rendering part, you will just have to rework the way it integrates with Vencord. Hope that it works out!

**Comment by javalsai** [comment]
> > :(
> > It's fine tho, I'll maybe end up writing myself a quick one when I get some time. Would really like to somehow manage to get autocompletion or view the rendered result as I've just learned LaTeX :P. I think discord should support LaTeX out of the box tho, the same way they have great support for codeblocks or file preview with syntax highlightning...
> 
> You should be able to reuse most of my code for the rendering part, you will just have to rework the way it integrates with Vencord. Hope that it works out!

I've heard about MathJax and KaTeX, so I'll try to integrate them as they seem really good options for LaTeX libraries. But I'll have to learn the Vencord API, so your code might come in handy, thanks!

But I can't see any diff on this PR for some reason, can I see it anywhere else?

Also, thanks for the advice!! <3

**Comment by jumpyjacko** [comment]
> I've heard about MathJax and KaTeX, so I'll try to integrate them as they seem really good options for LaTeX libraries. But I'll have to learn the Vencord API, so your code might come in handy, thanks!
> 
> But I can't see any diff on this PR for some reason, can I see it anywhere else?
> 
> Also, thanks for the advice!! <3

I almost didn't have the code because I deleted it all on my main computer, but luckily I have it pulled to my laptop. https://pastebin.com/svx87UKi

I chose not to use a library because I thought it would be unnecessary bloat but that's just my opinion so I used a web request since Discord is a web app anyways.

**Comment by javalsai** [comment]
> > I've heard about MathJax and KaTeX, so I'll try to integrate them as they seem really good options for LaTeX libraries. But I'll have to learn the Vencord API, so your code might come in handy, thanks!
> > But I can't see any diff on this PR for some reason, can I see it anywhere else?
> > Also, thanks for the advice!! <3
> 
> I almost didn't have the code because I deleted it all on my main computer, but luckily I have it pulled to my laptop. https://pastebin.com/svx87UKi
> 
> I chose not to use a library because I thought it would be unnecessary bloat but that's just my opinion so I used a web request since Discord is a web app anyways.
> 
> If you choose to use this code, I recommend taking a peek at the "/petpet" plugin and reference that since that was what I used as a reference for the Vencord APIs and my code is outdated.

Thanks!!!!

I see the point of doing web requests to avoid bloat and I'll consider it when I get some time to start the project.

I'll also take a look at `/petpet` as reference, surely will help me to get some Vencord API stuff.

Have a nice day!

**Pull request closed without merging**